### PR TITLE
SUIT-12569 Port some of the Watchme app tests to JS API

### PR DESCRIPTION
**Pull request by AntonSBilan**
@andriichumak @givehug what do you think should we use elements from element repo or better keep them locally?

### CODE REVIEW

**Comment by givehug** [comment]
> @andriichumak @givehug what do you think should we use elements from element repo or better keep them locally?

If there are some elements by default in initial WatchMe app, then yes, otherwise I'd use css selectors.

**Comment by andriichumak** [comment]
We can show-off both options. E.g. separate all element definitions to some file, have it there defined through api id, but at the end have a few elements defined with css selector and/or xpath with a comment: "You might want to define element selectors in code instead of Element Repository in case it is important for you to have branching and versioning for elements".

**Comment by andriichumak** [comment]
I can see in source that there are few elements already defined as css selector instead of API ID, so you can use those.

**Comment by givehug** [comment]
Do we need to add startTest and endTest commands similar how its implemented in current tests?

File: package.json [hunk 1]
```diff
@@ -12,6 +12,8 @@
     "interactive:debug": "suitest interactive --inspect-brk=9121 node ./node_modules/mocha/bin/_mocha --no-timeouts --exit \"./test/debug.test.js\"",
     "interactive:repl": "suitest interactive mocha --no-timeouts --exit \"./test/repl.test.js\"",
     "automated": "suitest automated mocha --no-timeouts --exit \"./test/dummy.test.js\"",
-    "automated:logs": "suitest automated --log-dir=e2e mocha --no-timeouts --exit \"./test/dummy.test.js\""
+    "automated:logs": "suitest automated --log-dir=e2e mocha --no-timeouts --exit \"./test/dummy.test.js\"",
+    "interactive:watchMe": "suitest interactive mocha --no-timeouts --exit ./watchMeTests/*.test.js",
```
[the remark this answers is not in the document]
Comment (AntonSBilan) [reply package.json]: Are you sure, did you test it? that should be evaluated as a string, almost everywhere we use it without extra quotes

[the remark this answers is not in the document]
Comment (givehug) [reply package.json]: No I did not. Evgen did it to other scripts.

[the remark this answers is not in the document]
Comment (AntonSBilan) [reply package.json]: In "interactive:debug" node arguments passed without quotes and I guess Nikola tested it on windows

[the remark this answers is not in the document]
Comment (givehug) [reply package.json]: Then lets remove these chars from other commands

[the remark this answers is not in the document]
Comment (AntonSBilan) [reply package.json]: @andriichumak do you have any objections about removing extra quotes from mocha arguments?

[the remark this answers is not in the document]
Comment (andriichumak) [reply package.json]: Let's remove it after testing on Window machine. If I remember it right, Windows does have some differences in treating string paths. E.g. if they include space or some special character. Not sure about *. Also, some terminals might interpolate glob patterns before running the command unless it's escaped, so I would be careful here.

File: watchMeTests/forwardingRewindingVideo.test.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+			{
+				name: PROP.VIDEO_URL,
+				val: 'http://file.suite.st/sampleapp/api/video/a84d67e5-c6d4-41c0-9f70-cd8c7a5f6f4f.mp4',
+			},
+		]).timeout(10000);
+		/**
+		 * Focus the "Forward" button on the control panel.
+		 */
+		await assert.press(VRC.DOWN);
+		await snapshotElement(pausedBtnFocused).timeout(2000);
```
Comment (andriichumak) [inline watchMeTests/forwardingRewindingVideo.test.js]: Timeout is 2s by default in JS API, so it's not really needed to specify it explicitly.

File: watchMeTests/navigationOnCategoryPages.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const {snapshotElement} = require('./utils');
+
+
+describe('Navigation on Category Pages', async() => {
+	before(async() => {
+		await assert.openApp();
+		await snippetHomepageOpened();
+	});
+
+	it(`Check if the navigation through the categories in the menu bar functions correctly
```
Comment (andriichumak) [inline watchMeTests/navigationOnCategoryPages.test.js]: I would keep `it` short, in this case maybe `it('Works as expected, ...`. And the text could go to comment line inside `it`.

File: watchMeTests/openImageInGallary.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const {snippetHomepageOpened} = require('./common');
+const suitest = require('suitest-js-api');
+const {assert, VRC, PROP, COMP, VALUE} = suitest;
+const {snapshotElement} = require('./utils');
+const {imageCandiesFocused, gallery, imageCandiesInGalleryRepo} = require('./elements');
+
+describe('Opening Image in Gallery', async() => {
```
Comment (andriichumak) [inline watchMeTests/openImageInGallary.test.js]: The idea of describe/it is that it should be human readable, i.e. "Describe image gallery. It should open correct image".
Please, look through test files and update in such fashion.

File: watchMeTests/returningToHomepage.test.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		 * Open the "Image Folder" and check that the image is focused.
+		 */
+		await assert.press(VRC.ENTER);
+		await snapshotElement(folderPathImages).timeout(2000);
+		await snapshotElement(imageInFolderChariesFocused);
+		/**
+		 * Return to the "Homepage" and check if the last element is focused.
+		 */
+		await assert.press(VRC.BACK);
+		await assert.element({
```
Comment (andriichumak) [inline watchMeTests/returningToHomepage.test.js]: Shouldn't it be part of the `elements.js`?

File: watchMeTests/forwardingRewindingVideo.test.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+	snippetHomepageOpened,
+	snippetOpenVideo} = require('./common');
+const {forwardBtnFocused, pausedBtnFocused, rewindBtnFocused} = require('./elements');
+const {snapshotElement} = require('./utils');
+
+const suitest = require('suitest-js-api');
+const {assert, VRC, PROP, COMP, VALUE, VIDEO_STATE} = suitest;
+
+describe('Forwarding/Rewinding video', async() => {
+	before(async() => {
```
Comment (andriichumak) [inline watchMeTests/forwardingRewindingVideo.test.js]: Each `before` should start with `startTest` command.

File: watchMeTests/forwardingRewindingVideo.test.js (hunk 2 above)
Comment (givehug) [reply watchMeTests/forwardingRewindingVideo.test.js]: > Timeout is 2s by default in JS API, so it's not really needed to specify it explicitly.

@andriichumak it actually was changed to 1ms `const DEFAULT_DELAY = 1;`. I'm not exactly sure why thought.

Comment (AntonSBilan) [reply watchMeTests/forwardingRewindingVideo.test.js]: With browser it works

Comment (andriichumak) [reply watchMeTests/forwardingRewindingVideo.test.js]: @givehug , delay for the button presses was changed, but not a timeout for assertions

Comment (givehug) [reply watchMeTests/forwardingRewindingVideo.test.js]: thats right

File: watchMeTests/elements.js [hunk 7]
```diff
@@ -0,0 +1,265 @@
+const suitest = require('suitest-js-api');
+const {PROP, COMP, VALUE} = suitest;
+
+module.exports = {
+	pausedBtnFocused: {
+		selector: '\'pause\' button focused',
```
Comment (JeKachan) [inline watchMeTests/elements.js]: strange selector :)

Comment (JeKachan) [reply watchMeTests/elements.js]: Is this property respresents apiId?

Comment (JeKachan) [reply watchMeTests/elements.js]: So, now I see what is it.

**Pull request merged**